charts/replicated-library/templates/lib/_labels.tpl

```diff
@@ -15,40 +15,40 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
     {{- end }}
   {{- end }}
 {{- end -}}
-
 {{/* Selector labels shared across objects */}}
 {{- define "replicated-library.labels.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "replicated-library.names.appname" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end -}}
-
 {{- define "replicated-library.labels.serviceSelectorLabels" -}}
   {{- $serviceValues := . -}}
-
   {{- if hasKey . "ObjectValues" -}}
     {{- with .ObjectValues.values -}}
       {{- $serviceValues = . -}}
     {{- end -}}
   {{ end -}}
-
-  {{- if $serviceValues.appName }}
-    {{- $matchingAppFound := false -}}
-
-    {{- range $appName, $appValues := .Values.apps }}
-      {{- if and $appValues.enabled (eq $appName $serviceValues.appName) (ne $matchingAppFound true) -}}
-        {{- $matchingAppFound = true -}}
-app.kubernetes.io/name: {{ $appName }}
+  {{- if $serviceValues.selector -}}
+{{ toYaml $serviceValues.selector }}
+  {{- else -}}
+    {{- if $serviceValues.appName }}
+      {{- range $serviceValues.appName }}
```

Comment:
While testing this, I've realized you've changed this from a string to a list which is a braking change. We're early in the chart so we can still decide to do breaking changes if we want but I'm not sure if you had through that through and intended to make a case for that.

Range doesn't appear to work on type string so the current method of expecting a single string fails on range.

After updating my test chart to have a list (of one item) I've found a failure on the Statefulset where a comparison to appName is done which again was expecting a string and not a list. So we just need to decide which way we want to go with this and test/update the rest of the templates that reference this field.